test/mocks/index.ts

```diff
@@ -0,0 +1,218 @@
+    BigNumber,
+    BigNumber,
+    BigNumber,
+    BigNumber,
+    BigNumber,
+    string,
+    BigNumber,
+    BigNumber,
+    BigNumber
+  ] & {
+    sellerWallet: string;
+    parentDomainId: BigNumber;
+    salePrice: BigNumber;
+    privateSalePrice: BigNumber;
+    mintlistSaleDuration: BigNumber;
+    amountForSale: BigNumber;
+    mintlistMerkleRoot: string;
+    startingMetadataIndex: BigNumber;
+    folderGroupID: BigNumber;
+    publicSaleLimit: BigNumber;
+  };
+  currentSalePhase: number;
+  currentDomainsSold: BigNumber;
+  currentSaleStartTimestamp: BigNumber;
+  currentLatestBlock: Partial<Block>;
+  currentSaleId: string;
+  currentSaleCounter: BigNumber;
+  currentPauseStatus: boolean;
+  currentSignerFunds: ethers.BigNumber;
+  currentDomainsPurchasedByAccountPerSale: BigNumber;
+  reset: () => void;
+};
+
+export function getDefaultMockSaleConfiguration(): 
+  [
+    string,
+    BigNumber,
+    BigNumber,
+    BigNumber,
+    BigNumber,
+    BigNumber,
+    string,
+    BigNumber,
+    BigNumber,
+    BigNumber
+  ] & {
+    sellerWallet: string;
+    parentDomainId: BigNumber;
+    salePrice: BigNumber;
+    privateSalePrice: BigNumber;
+    mintlistSaleDuration: BigNumber;
+    amountForSale: BigNumber;
+    mintlistMerkleRoot: string;
+    startingMetadataIndex: BigNumber;
+    folderGroupID: BigNumber;
+    publicSaleLimit: BigNumber;
+  }
+ {
+  return {
+    sellerWallet: "0x1234567890123456789012345678901234567890",
+    parentDomainId: BigNumber.from(123),
+    salePrice: ethers.utils.parseEther("1.0"), // 1 ^ 18th
+    privateSalePrice: ethers.utils.parseEther("2.0"), // 2 ^ 18th
+    mintlistSaleDuration: BigNumber.from(3600), // ~ 1hr (seconds)
+    amountForSale: BigNumber.from(50),
+    mintlistMerkleRoot: "0xabcdef1234567890abcdef1234567890abcdef12",
+    startingMetadataIndex: BigNumber.from(1),
+    folderGroupID: BigNumber.from(2),
+    publicSaleLimit: BigNumber.from(10),
+  } as any;
+}
+
+export let mocks: Mocks = {
+  currentSaleConfiguration: getDefaultMockSaleConfiguration(),
+  currentSalePhase: SalePhase.Private,
+  currentDomainsSold: BigNumber.from(0),
+  currentSaleStartTimestamp: BigNumber.from(feb23TsSeconds), // ~2/23/2023
+  currentLatestBlock: { timestamp: feb24TsSeconds }, // ~2/24/2023
+  currentSaleId: "12345",
+  currentSaleCounter: BigNumber.from(1),
+  currentPauseStatus: false,
+  currentSignerFunds: ethers.utils.parseEther("10.0"),
+  currentDomainsPurchasedByAccountPerSale: BigNumber.from(0),
+
+  signer: {
+    getAddress: async () => "0x1234567890123456789012345678901234567890",
+    getBalance: async () => mocks.currentSignerFunds,
+  },
+
+  sale: {
+    provider: {
+      async getBlock(): Promise<Block> {
+        return Promise.resolve(mocks.currentLatestBlock as Block);
+      },
+    } as unknown as Provider,
+
+    async salePhase(): Promise<number> {
+      return mocks.currentSalePhase;
+    },
+
+    async saleConfiguration(overrides?: CallOverrides): Promise<
+      [
```

Comment:
Again try to see if you can do something that's less verbose. This type is used in several places but requires 20+ lines of code. I think you should be able to do `Promise<Mocks.SaleConfig>` as a return type instead